--- kicad-4.0.6/CMakeLists.txt~	2017-04-15 19:08:53.000000000 +0200
+++ kicad-4.0.6/CMakeLists.txt	2017-04-15 21:09:33.888727818 +0200
@@ -359,7 +359,7 @@
         set( KICAD_PLUGINS ${KICAD_BIN}/scripting/plugins
             CACHE PATH "Location of KiCad plugins." )
     else()
-        set( KICAD_PLUGINS lib/kicad/plugins
+        set( KICAD_PLUGINS lib${LIB_SUFFIX}/kicad/plugins
             CACHE PATH "Location of KiCad plugins." )
     endif()
 
