diff -uNr kicad.orig/3d-viewer/CMakeLists.txt kicad/3d-viewer/CMakeLists.txt
--- kicad.orig/3d-viewer/CMakeLists.txt	2012-01-31 08:43:32.000000000 +0100
+++ kicad/3d-viewer/CMakeLists.txt	2012-02-07 22:56:39.427930449 +0100
@@ -19,4 +19,4 @@
     trackball.cpp
     )
 
-add_library(3d-viewer ${3D-VIEWER_SRCS})
+add_library(3d-viewer STATIC ${3D-VIEWER_SRCS})
diff -uNr kicad.orig/bitmaps_png/CMakeLists.txt kicad/bitmaps_png/CMakeLists.txt
--- kicad.orig/bitmaps_png/CMakeLists.txt	2012-01-31 08:43:32.000000000 +0100
+++ kicad/bitmaps_png/CMakeLists.txt	2012-02-07 22:56:56.861533429 +0100
@@ -676,5 +676,5 @@
 endforeach()
 
 #add_library( bitmaps SHARED ${CPP_LIST} )
-add_library( bitmaps ${CPP_LIST} )
+add_library( bitmaps STATIC ${CPP_LIST} )
 
diff -uNr kicad.orig/bitmaps_xpm/CMakeLists.txt kicad/bitmaps_xpm/CMakeLists.txt
--- kicad.orig/bitmaps_xpm/CMakeLists.txt	2012-01-31 08:43:32.000000000 +0100
+++ kicad/bitmaps_xpm/CMakeLists.txt	2012-02-07 22:57:15.545155677 +0100
@@ -449,4 +449,4 @@
     set_source_files_properties(${CPP_BITMAP} PROPERTIES COMPILE_FLAGS -DXPMMAIN)
 endforeach(LOOP_VAR)
 
-add_library(bitmaps ${CPP_BITMAPS})
+add_library(bitmaps STATIC ${CPP_BITMAPS})
diff -uNr kicad.orig/common/CMakeLists.txt kicad/common/CMakeLists.txt
--- kicad.orig/common/CMakeLists.txt	2012-01-31 08:43:32.000000000 +0100
+++ kicad/common/CMakeLists.txt	2012-02-07 22:57:43.145582514 +0100
@@ -77,7 +77,7 @@
     zoom.cpp
     )
 
-add_library(common ${COMMON_SRCS})
+add_library(common STATIC ${COMMON_SRCS})
 
 set(PCB_COMMON_SRCS
     base_screen.cpp
@@ -125,7 +125,7 @@
     COMPILE_DEFINITIONS "PCBNEW"
     )
 
-add_library(pcbcommon ${PCB_COMMON_SRCS})
+add_library(pcbcommon STATIC ${PCB_COMMON_SRCS})
 
 
 # auto-generate netlist_lexer.h and netlist_keywords.cpp
diff -uNr kicad.orig/polygon/CMakeLists.txt kicad/polygon/CMakeLists.txt
--- kicad.orig/polygon/CMakeLists.txt	2012-01-31 08:43:32.000000000 +0100
+++ kicad/polygon/CMakeLists.txt	2012-02-07 22:57:06.361680317 +0100
@@ -10,4 +10,4 @@
     polygon_test_point_inside.cpp
     )
 
-add_library(polygon ${POLYGON_SRCS})
+add_library(polygon STATIC ${POLYGON_SRCS})
diff -uNr kicad.orig/polygon/kbool/src/CMakeLists.txt kicad/polygon/kbool/src/CMakeLists.txt
--- kicad.orig/polygon/kbool/src/CMakeLists.txt	2012-01-31 08:43:32.000000000 +0100
+++ kicad/polygon/kbool/src/CMakeLists.txt	2012-02-07 22:57:55.352437924 +0100
@@ -18,4 +18,4 @@
     record.cpp
     scanbeam.cpp)
 
-add_library(kbool ${KBOOL_SRCS})
+add_library(kbool STATIC ${KBOOL_SRCS})
diff -uNr kicad.orig/potrace/CMakeLists.txt kicad/potrace/CMakeLists.txt
--- kicad.orig/potrace/CMakeLists.txt	2012-01-31 08:43:32.000000000 +0100
+++ kicad/potrace/CMakeLists.txt	2012-02-07 22:57:23.878617863 +0100
@@ -13,4 +13,4 @@
     trace.cpp
     )
 
-add_library(potrace ${POTRACE_SRCS})
+add_library(potrace STATIC ${POTRACE_SRCS})
