--- kicad/makefile.gtk.org	2008-09-27 16:28:55.826173110 +0200
+++ kicad/makefile.gtk	2008-09-27 16:30:43.141448464 +0200
@@ -2,8 +2,8 @@
 MAKEGTK = $(MAKE) -f makefile.gtk
 KICAD_SUBDIRS = common bitmaps 3d-viewer polygon polygon/kbool/src pcbnew eeschema eeschema/plugins cvpcb kicad gerbview
 KICAD_SUBDIRS_BIN = eeschema eeschema/plugins pcbnew cvpcb kicad gerbview
-KICAD_SUBDIRS_RES = internat modules template library
-KICAD_SUBDIRS_HELP = help
+KICAD_SUBDIRS_RES = internat template
+KICAD_SUBDIRS_HELP = 
 
 all:
 	@for dir in $(KICAD_SUBDIRS); do \
